doc/spec_TransferElement.dox

```diff
@@ -138,48 +138,53 @@ This documnent is currently still **INCOMPLETE**!
       - 8.5.4 If Device::activateAsyncRead() is called while the device is not opened or has an error, this call has no effect. If it is called when no deactivated transfer element exists, this call also has no effect.
       - 8.5.5 Device::activateAsyncRead() does not throw any exceptions.
       - 8.5.6 When Device::activateAsyncRead() returns, it is not guaranteed that all initial values have been received already.
+      - \anchor transferElement_B_8_5_7 8.5.7 Each backend is responsible for activating asynchronous reads only on those TransferElements which it created.
+              Meta-backends like the LogicalNameMappingBackend delegate Device::activateAsyncRead() to all of their target backends, as they also delegate the creation of
+              the TransferElements to their targets. (cf. \ref transferElement_B_9_2_1 "9.2.1")
   - \anchor transferElement_B_8_6 8.6 Blocking TransferElement::read() calls can be interrupted by the application via TransferElement::interrupt(). [\ref testTransferElement_B_8_6 "T" (with sub-points), \ref UnifiedTest_TransferElement_B_8_6 "U" (with sub-points)]
-    - \anchor transferElement_B_8_6_1 8.6.1 TransferElement::interrupt() places a boost::thread_interrupted exception into the TransferElement::_readQueue.
+    - \anchor transferElement_B_8_6_1 8.6.1 TransferElement::interrupt() places a boost::thread_interrupted exception into the \ref transferElement_data_transport_queue "implementation-specific data transport queue". [\ref testTransferElement_B_8_6_1 "T"]
     - \anchor transferElement_B_8_6_2 8.6.2 This casues any read operation to complete immediately and throw the boost::thread_interrupted exception, after any unread data on the queue before the exception has been read normally.
     - \anchor transferElement_B_8_6_3 8.6.3 The operation receiving the boost::thread_interrupted exception will also obey \ref transferElement_B_5 "5", i.e. postRead() is still called.
     - \anchor transferElement_B_8_6_4 8.6.4 Calling TransferElement::interrupt() does not disturb the subsequent operation of the TransferElement.
     - \anchor transferElement_B_8_6_5 8.6.5 Calling TransferElement::interrupt() without AccessMode::wait_for_new_data will cause a ChimeraTK::logic_error.
+    - \anchor transferElement_B_8_6_6 8.6.6 TransferElement::interrupt() is a virtual function which has to be implemented by all TransferElement implementations \ref transferElement_comment_B_8_6_6 "(*)". This should be done by calling TransferElement::interrupt_impl with the \ref transferElement_data_transport_queue "implementation-specific data transport queue" as argument.
```

Comment:
I am not sure about the "must be". As we don't necessarily have access to the queue in decorator-like implementations (also the FeedingFanOut and the BiDirectionalProcessArray, for instance). They have to delegate.

Do we have to be more explicit about decorators vs. "full" implementations.

Reply:
right, decorators must delegate, all others must call that implementation in the base class.

Reply:
OK, so it's two times must.

But the comment stays valid: as soon as the queue is known in NDRegisterAccessor, and decorators set it like readQueue, the virtual can go away and it can be implemented in the base class. We need a getter for it as well for the decorators, but that's an implementation detail.